src/object/moving_sprite.cpp

```diff
@@ -230,5 +230,24 @@ MovingSprite::spawn_explosion_sprites(int count, const std::string& sprite_path)
                                              LAYER_OBJECTS-1);
   }
 }
+void
+MovingSprite::spawn_flame_sprites(int count, const std::string& sprite_path,Color color_)
+{
+    for (int i = 0; i < count; i++) {
+      Vector ppos = m_col.m_bbox.get_middle();
+      float angle = angle;
+      float velocity = 0;
+      float vx = sinf(angle)*velocity;
+      float vy = -cosf(angle)*velocity;
+      Vector pspeed = Vector(vx, vy);
+      Vector paccel = Vector(0, Sector::get().get_gravity()*10);
+      Sector::get().add<SpriteParticle>(sprite_path,
+                                             "default",
+                                             ppos, ANCHOR_MIDDLE,
+                                             pspeed, paccel,
+                                             LAYER_OBJECTS-1,false,color_);
+  }
+}
```

Comment:
Two things about this function:
1. It shouldn't be here. It should be in the Flame class because it's related to the flame.
2. What do you mean by `float angle = angle;`???